src/components/checkBoxSelection/checkBoxSelection.js

```diff
@@ -1,47 +1,99 @@
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
-import './checkBoxSelection.scss';
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import "./checkBoxSelection.scss";
+import { connect } from "react-redux";
+import {
+  ADD_VARIABLE,
+  UPDATE_VARIABLE,
+  DELETE_VARIABLE
+} from "../../store/actionTypes";
+
 class CheckBoxSelection extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            isClicked: false
-        };
+  constructor(props) {
+    super(props);
+    this.state = {
+      isClicked: this.props.isSelected
+    };
+  }
+
+  convertSvgToDataUrl = file => {
+    let reader = new FileReader();
+    reader.onload = result => {
+      this.props.svgSettingList.push({
+        name: file.name,
+        dataurl: result.target.result
+      });
+    };
+    if (file) {
+      reader.readAsDataURL(file);
     }
+  };
 
-    componentWillMount() {
-	 this.setState({ isClicked: this.props.isSelected }); 
+  changeColourAndTick = () => {
+    let index = 0;
+    this.setState({ isClicked: !this.state.isClicked });
+    index = this.props.svgSettingList.findIndex(
+      setting => setting.name == this.props.file.name
+    );
+
+    if (index < 0) {
+      this.convertSvgToDataUrl(this.props.file);
+    } else {
+      this.props.svgSettingList.splice(index, 1);
     }
+  };
 
-	handleDivClick= () => {
-	    this.setState({
-	        isClicked: !this.state.isClicked
-	    });
-	}
-
-	render() {
-
-	    return (
-
-	        <div
-    		className={this.state.isClicked ? 'checkbox-container-clicked' : 'checkbox-container-unclicked'}
-    		onClick={this.handleDivClick}
-	        >
-	            <input
-    			className='checkbox-input'
-    			type='checkbox'
-    			checked={this.state.isClicked}
-	            />
-	            <label className='checkbox-label'>
-					  {this.props.filename}
-	            </label>
-	        </div>
-	    );
-	}
+  render() {
+    return (
+      <div
+        className={
+          this.state.isClicked
+            ? "checkbox-container-clicked"
+            : "checkbox-container-unclicked"
+        }
+        onClick={this.changeColourAndTick}
```

Comment:
it would be better to add () => before this.changeColourAndTick(), this is to make sure that the context of "this" is correct for this call. Sometimes the context of "this" can be obscured in react, and leading with an anonymous function can help with this assignment () =>

Reply:
this applies to any function you try to call from within a render

Reply:
I have already pushed ()=> implementation.